crates/turborepo-globwalk/src/lib.rs

```diff
@@ -277,6 +284,86 @@ fn collapse_path(path: &str) -> Option<(Cow<str>, usize)> {
     }
 }
 
+pub fn globwalk(
+    base_path: &AbsoluteSystemPath,
+    include: &[String],
+    exclude: &[String],
+    walk_type: WalkType,
+) -> Result<HashSet<AbsoluteSystemPathBuf>, WalkError> {
+    let (base_path_new, include_paths, exclude_paths) =
+        preprocess_paths_and_globs(base_path, include, exclude)?;
+    let inc_patterns = include_paths
+        .iter()
+        .map(|g| {
+            Glob::new(g.as_str()).map_err(|e| e.into()).and_then(|g| {
+                if g.has_root() {
+                    Ok(g)
+                } else {
+                    Err(WalkError::InternalError {
+                        glob: g.to_string(),
+                        error: "expected glob to have a root".to_string(),
+                    })
+                }
+            })
```

Comment:
```suggestion
            match Glob::new(g.as_str()) {
                Err(e) => e.into(),
                Ok(g) if g.has_root() => Ok(g),
                Ok(g) => Err(WalkError::InternalError {
                    glob: g.to_string(),
                    error: "expected glob to have a root".to_string(),
                })
            }
```

Reply:
This is sadly irrelevant in the latest revision, we can't rely on globs being rooted, as they don't count as rooted on windows.